test/kafo/hook_context_test.rb

```diff
@@ -85,5 +86,23 @@ module Kafo
         assert_equal false, context.app_option?('unknown_option')
       end
     end
+
+    describe "#has_custom_fact? with fact value present" do
+      let(:config) { Minitest::Mock.new }
+      let(:custom_fact_storage) { { 'my_custom_fact' => 'my_custom_fact_value' } }
+      kafo.expect :config, config
+      config.expect :custom_fact_storage, custom_fact_storage
+      assert_equal true, context.has_custom_fact?('my_custom_fact')
+      assert_equal false, context.has_custom_fact?('not_my_custom_fact')
```

Comment:
I think you want to wrap the 4 lines above this point in a `specify do` block. Same with the next test you added.